chainer/optimizer.py

```diff
@@ -369,6 +369,19 @@ class GradientMethod(Optimizer):
 
     """
 
+    def _replace_cleared_grads(self):
+        for name, param in self.target.namedparams():
+            if param.grad is None:
+                with cuda.get_device(param.data):
+                    xp = cuda.get_array_module(param.data)
+                    param.grad = xp.zeros_like(param.data)
+
```

Comment:
Can you make this method public? By doing so, derived optimizers can override it to skip reallocation if not needed. (See the TODO below)
```
# TODO(unno): Some optimizers can skip this process if they does not
# affect to a parameter when its gradient is zero.
```

Then you can also remove this TODO.

As for the method name, `_replace_cleared_grads()` can be a bit ambiguous as a public method. Can you rename it to `reallocate_cleared_grads()`?